pkg/controllers/jobset_controller.go

```diff
@@ -196,30 +231,29 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 	ctx = ctrl.LoggerInto(ctx, log)
 
 	for _, rjob := range js.Spec.Jobs {
-		job, err := r.constructJobFromTemplate(js, &rjob)
+		jobs, err := r.constructJobsFromTemplate(js, &rjob, ownedJobs)
 		if err != nil {
 			return err
 		}
 
-		// Check if we need to create this job.
-		// If not, skip this job and continue iterating to the next job.
-		if create := r.shouldCreateJob(ctx, job, ownedJobs); !create {
-			continue
-		}
+		for _, job := range jobs {
+			// Create headless service if specified for this job.
+			if dnsHostnamesEnabled(&rjob) {
+				if !isIndexedJob(job) {
```

Comment:
I see you already added the check here, I guess the integration test is what is left, right?